resource/packages/apt/apt.go

```diff
@@ -0,0 +1,88 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apt
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/asteris-llc/converge/resource"
+)
+
+// Package is an API for package state
+type Package struct {
+	resource.TaskStatus
+
+	Name  string
+	State string
+}
+
+// Check if the package has to be 'installed', or 'absent'
+func (p *Package) Check(resource.Renderer) (resource.TaskStatus, error) {
+	status := resource.NewStatus()
+
+	currentPkgStatusRaw, err := exec.Command("bash", "-c", "apt-cache policy "+p.Name+" | grep Installed | awk '{print $2}'").Output()
+	if err != nil {
+		log.WithField("module", "apt").Warn(err)
```

Comment:
this should return the error instead of logging it and continuing. Maybe this:

```go
return status, errors.Wrapf(err, "checking package %s", p.Name)
```